### PR TITLE
Package cow.2.3.0

### PR DESCRIPTION
**Pull request by samoht**
### `cow.2.3.0`

Caml-On-the-Web: XML, JSON, HTML and Markdown libraries

Writing web-applications requires a lot of skills: HTML, XML, JSON and
Markdown, to name but a few!  This library provides OCaml combinators
for these web formats by:

See more explanation at: http://mirage.github.io/ocaml-cow

This library is in beta, and full documentation is still being written.
Some repositories which use it include:

* Mirage website: http://github.com/mirage/mirage-www
* Opam2web: http://github.com/OCamlpro/opam2web

---
* Homepage: https://github.com/mirage/ocaml-cow
* Source repo: https://github.com/mirage/ocaml-cow.git
* Bug tracker: https://github.com/mirage/ocaml-cow/issues

---

---
### 2.3.0 (29-04-2018)

* Port to jbuilder/dune (@samoht, #102)
* Fix atom feeds (@hannesm, #101)
* Improve the functions generating HTML tags with no content (@Chris00, #100)
* Improve documentation (@Chris00, #99)
* Add a "ty" (type) argument to `Html.link` (@Chris00, #99)
* Update `Html.script` to take a Uri.t for "src" (@Chris00, #99)
* `Html.a`: make ~href optional (also used for anchors) (@Chris00, #99)
* Add the optional argument ?attrs to `Html.a` (@Chris00, #99)
* Use <thead> and <tbody> for table construction (when appropriate) (@Chris00, #99)
:camel: Pull-request generated by opam-publish v0.3.5

### CODE REVIEW

File: packages/cow/cow.2.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          "Rudi Grinberg" "Timothy Bourke"]
+homepage: "https://github.com/mirage/ocaml-cow"
+doc: "http://mirage.github.io/ocaml-cow/"
+license: "ISC"
+dev-repo: "https://github.com/mirage/ocaml-cow.git"
+bug-reports: "https://github.com/mirage/ocaml-cow/issues"
+tags: ["org:mirage" "org:xapi-project" "www" "html" "xml"
+       "css" "json" "markdown" ]
+
+available: [ ocaml-version >= "4.02.3"]
```
Comment (mseri) [inline packages/cow/cow.2.3.0/opam]: There is a failure in the CI due to the use of String.lowercase_ascii that was introduced in 4.03

File: packages/cowabloga/cowabloga.0.2.2/opam [hunk 2]
```diff
@@ -10,7 +10,7 @@ build: [make "all"]
 install: [make "install"]
 remove: ["ocamlfind" "remove" "cowabloga"]
 depends: [
-  "cow" {>= "2.0.0"}
+  "cow" {>= "2.0.0" & < "2.3.0"}
   "omd" {>= "0.8.2"}
   "lwt" {>= "2.4.3"}
   "cohttp" {>= "0.15.0"}
```
Comment (mseri) [inline packages/cowabloga/cowabloga.0.2.2/opam]: Given that you are touching these can you also add < 1.0.0 bounds on cohttp dependencies in cowabloga?

**Comment by samoht** [comment]
I would like to fix the other revdeps: I will do it in another PR to not overload this one, but I will rebase it regularly. So please don't merge until everything is green :-)

**Comment by samoht** [comment]
Rebased. Hopefully the revdeps will be mostly green and this can be merged.

**Comment by samoht** [comment]
There's an issue with conduit that I would like to fix. Please do not merge yet.

**Comment by samoht** [comment]
Conduit issues should be fixed by https://github.com/ocaml/opam-repository/pull/11952

**Comment by samoht** [comment]
lwt_react issues should be fixed by https://github.com/ocaml/opam-repository/pull/11954

**Comment by samoht** [comment]
ocaml-git issues should be fixed by https://github.com/ocaml/opam-repository/pull/11955

**Comment by samoht** [comment]
The websocket issue should be fixed by https://github.com/ocaml/opam-repository/pull/11956

**Comment by samoht** [comment]
More cohttp test fixes in https://github.com/ocaml/opam-repository/pull/11970

**Comment by samoht** [comment]
I think this is ready to merge. The remaining error is due to a solver timeout:

```
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-family=debian
[ERROR] Sorry, resolution of the request timed out.
        Try to specify a simpler request, use a different solver, or increase the allowed time by setting OPAMSOLVERTIMEOUT to a bigger value (currently, it is set to 60.0 seconds).
Command failed: opam list --readonly --external  --resolve=cowabloga.0.2.2 returned 60
```

@AltGr anything you would like to gather to maybe reproduce/fix?

**Comment by mseri** [comment]
Thanks for this monstrous amount of fixes! I think we can merge

**Pull request merged**